### ISSUE TITLE
Adding node to build images

### ISSUE THREAD

**Issue by wtgodbe** [dotnet/dnceng#989]
dotnet/aspnetcore installs node as part of its CI build: https://github.com/dotnet/aspnetcore/blob/a2ce0b2bac64c03b4efc76de59fd5b9de4469b5a/.azure/pipelines/jobs/default-build.yml#L200-L204. Recently, we've been seeing more transient failures than usual in that step. Would it be possible to get node/npm pre-installed on some of our build images so that we can increase our build reliability? We currently use 18.x, and try to stay on the newest active LTS, so it'd be great to have LTS on the images we use (both Windows & non-Windows)

**Comment by amcasey** [comment]
Our (aspnetcore) last two rolling builds failed to download node 18 (after 5 minutes, they had half the bytes).

**Comment by garath** [comment]
Since these failures are from the `NodeTool` task, it is probably worth pursuing an issue with Azure DevOps about reliability problems.

**Comment by garath** [comment]
Dnceng does have some telemetry on these instances. Here's a quick graph I made:

Top graph is "failed" instances, bottom graph is "succeeded" instances, both are over the last 90 days.

<img width="704" alt="image" src="https://github.com/dotnet/dnceng/assets/2303549/002eac4b-c9a4-4e9d-bbf2-ce658526f958">

Execute in [[Web](https://dataexplorer.azure.com/clusters/engsrvprod/databases/engineeringdata?query=H4sIAAAAAAAAAz3NOw7CMBAE0J5TDJUdKQWUNBwhRfAFNvYKW3HsyB9FIA6PkZC7ld7MrHIbexd4Zh2TyacPDsuJ8SiUimqIO+gZ5e1iho6K8jpRM0sZYoqGVYxedJ85V19wdgFS5NXtOxsxQmgKmn27f1O5bhsl92boWEORA5YXFhdkfz3iaobxv9YaiYPhhNJI25b5AlOzMAm8AAAA)] [[Desktop](https://engsrvprod.kusto.windows.net/engineeringdata?query=H4sIAAAAAAAAAz3NOw7CMBAE0J5TDJUdKQWUNBwhRfAFNvYKW3HsyB9FIA6PkZC7ld7MrHIbexd4Zh2TyacPDsuJ8SiUimqIO+gZ5e1iho6K8jpRM0sZYoqGVYxedJ85V19wdgFS5NXtOxsxQmgKmn27f1O5bhsl92boWEORA5YXFhdkfz3iaobxv9YaiYPhhNJI25b5AlOzMAm8AAAA&web=0)] [[cluster('engsrvprod.kusto.windows.net').database('engineeringdata')](https://dataexplorer.azure.com/clusters/engsrvprod/databases/engineeringdata)]

```kql
TimelineRecords
| where StartTime > ago(90d)
| where TaskName has 'NodeTool'
| where Result !in ('skipped', 'canceled')
| summarize count() by bin(StartTime, 1d), Result
| render timechart
```

**Comment by garath** [comment]
> Our (aspnetcore) last two rolling builds failed to download node 18 (after 5 minutes, they had half the bytes).

@amcasey link the build, please?

**Comment by amcasey** [comment]
@garath I was torn about whether that would be helpful. 😆 

https://dev.azure.com/dnceng-public/public/_build/results?buildId=449442&view=results
https://dev.azure.com/dnceng-public/public/_build/results?buildId=449541&view=results

**Comment by garath** [comment]
Ah thanks. I just wanted to check that the failures were showing in the query I made above. They are.

**Comment by wtgodbe** [comment]
> Since these failures are from the `NodeTool` task, it is probably worth pursuing an issue with Azure DevOps about reliability problems.

@garath do you have a recommendation on how to do that? Do we have any internal way to file issues with the team?

**Comment by garath** [comment]
Azure DevOps makes it nice and easy. There's a little circled question mark next to your profile picture that holds all of the help/docs/support links. The "Contact Support" link will send you to an internal wiki with all of the options (including ICM).

**Comment by wtgodbe** [comment]
https://portal.microsofticm.com/imp/v3/incidents/incident/436333314/summary

**Comment by wtgodbe** [comment]
@garath looks like the azure folks don't plan on working to improve the reliability of the task, since what it downloads is a 3rd party asset. Is there any way we could get node added to the images we use?

> You are using 1ES hosted pool. Create your own image with the right nodejs version and you will eliminate a step that will always have a chance to fail.

**Comment by garath** [comment]
Yes, this can be added to the machines. It's already in the backlog.

**Comment by ilyas1974** [comment]
As the priority of this has not increased in over a year, we do not have the resources to address this currently. Should we be in error, please reopen the issue
